### PR TITLE
plugin-hooks-store

### PR DESCRIPTION
**Pull request by sdu-gyf**
it works on my machine :)

how to run it:
1. `cd packages/plugin-hooks-store`
2. `npm install`
3. `npm build`
4. `cd ../../examples/basic-hooks-store && npm install`
5. `npm link ../../packages/plugin-hooks-store`
6. `npm run start`

I hope these codes will work as expected on your computer too

### CODE REVIEW

**Comment by sdu-gyf** [comment]
todo:
- [ ] 解决和 ice store 混用时包两个 Provider 的情况

File: packages/plugin-hooks-store/package.json [hunk 1]
```diff
@@ -5,19 +5,24 @@
   "main": "lib/index.js",
   "scripts": {
     "start": "tsc -w",
-    "build": "rm -rf lib && tsc"
+    "build": "rm -rf lib && tsc && cp -R ./src/template ./lib/template"
```
Comment (luhc228) [inline packages/plugin-hooks-store/package.json:8]: 不需要 `cp -R ./src/template ./lib/template`

`npm run watch` 或者 `npm run bootstrap` 的时候会把模板复制过去

File: packages/plugin-hooks-store/src/template/pageComponent.tsx.ejs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<% if(!hasJsxRuntime) { %>
```
Comment (luhc228) [inline packages/plugin-hooks-store/src/template/pageComponent.tsx.ejs:1]: 有传这个变量进来吗？

File: packages/plugin-hooks-store/src/generator.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
+import * as path from 'path';
+import * as fse from 'fs-extra';
+import * as recursiveReaddir from 'fs-readdir-recursive';
+import checkPageIndexFileExists from './utils/checkPageIndexFileExists';
+import { ProjectType } from './utils/types';
+
```
Comment (luhc228) [inline packages/plugin-hooks-store/src/generator.ts:6]: 格式

File: packages/plugin-hooks-store/src/generator.ts [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    };
+  }
+
+  private renderPageHooksStore({ pageName, pageNameDir, pageHooksDir, pageHooksFile, existedPageHooksStoreFile }: IRenderPageParams) {
+    if (!existedPageHooksStoreFile && (fse.pathExistsSync(pageHooksDir) || fse.pathExistsSync(pageHooksFile))) {
+      const sourceFilename = 'hooksStore';
+      const exportName = 'hooksStore';
+      const targetPath = path.join(this.targetPath, 'pages', pageName, `${sourceFilename}.ts`);
+
+      const pageHookFilePath = path.join('use', pageNameDir);
```
Comment (luhc228) [inline packages/plugin-hooks-store/src/generator.ts:144]: `'use'`?

File: packages/plugin-hooks-store/src/template/pageHooksStore.ts.ejs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+const hook = { default: <%- hooksStr %> }
+const store = createStore(hook);
+export default store;
+<% } %>
+
+<% if (importStr && !isSingleHook) { %>
+const hooks = { <%- hooksStr %> }
+const hooksStore = createStore(hooks);
+export default hooksStore;
+<% } %>
```
Comment (luhc228) [inline packages/plugin-hooks-store/src/template/pageHooksStore.ts.ejs:16]: 这里两段逻辑有点重复了，可以考虑简化一下。至少这两段可以是公共的
```js
const hooksStore = createStore(hooks);
export default hooksStore;
```

File: packages/plugin-hooks-store/package.json (hunk 1 above)
Comment (sdu-gyf) [reply packages/plugin-hooks-store/package.json]: solved [fb9fe29](https://github.com/alibaba/ice/pull/4121/commits/fb9fe298ca1192138209e7df6ca1a4aa39da6b7b)

File: packages/plugin-hooks-store/src/template/pageComponent.tsx.ejs (hunk 2 above)
Comment (sdu-gyf) [reply packages/plugin-hooks-store/src/template/pageComponent.tsx.ejs]: solved [7d677e6](https://github.com/alibaba/ice/pull/4121/commits/7d677e68f46a05d52dec71e2d87c92b1af24fbb0)

File: packages/plugin-hooks-store/src/generator.ts (hunk 3 above)
Comment (sdu-gyf) [reply packages/plugin-hooks-store/src/generator.ts]: solved [d36ddbb](https://github.com/alibaba/ice/pull/4121/commits/d36ddbb8d79e76c889b9f0b9ca4478a1c192f87b)

File: packages/plugin-hooks-store/src/generator.ts (hunk 4 above)
Comment (sdu-gyf) [reply packages/plugin-hooks-store/src/generator.ts]: solved [d36ddbb](https://github.com/alibaba/ice/pull/4121/commits/d36ddbb8d79e76c889b9f0b9ca4478a1c192f87b)

File: packages/plugin-hooks-store/src/template/pageHooksStore.ts.ejs (hunk 5 above)
Comment (sdu-gyf) [reply packages/plugin-hooks-store/src/template/pageHooksStore.ts.ejs]: solved [7d677e6](https://github.com/alibaba/ice/pull/4121/commits/7d677e68f46a05d52dec71e2d87c92b1af24fbb0)

**Comment by alvinhui** [comment]
#4102

**Pull request closed without merging**